### ISSUE TITLE
`W505` reported as `E501` a.k.a doc-line-too-long

### ISSUE THREAD

**Issue by Borda** [astral-sh/ruff#11100]
it could be just my wrong understanding or expectations but running formatting on https://github.com/unit8co/darts/pull/2341, and they would like to make 88 line-length for code only and docstring differently, so I found [`W505`](https://docs.astral.sh/ruff/rules/doc-line-too-long/)
which I wanted to ignore, but it turned out it is not reported but `E501`. Well, `E501` is correct, but it is a superset, and `W505` would better describe the case...

using:
```yaml
  - repo: https://github.com/astral-sh/ruff-pre-commit
    rev: v0.4.1
    hooks:
      - id: ruff
```

**Comment by charliermarsh** [comment]
Yes I believe it's intentional that we report E501 on any overlong lines, and W505 on overlong lines in documentation. I think this is by design. What's the problem that it's causing?

**Comment by Borda** [comment]
> I think this is by design. What's the problem that it's causing?

`W505` is expected but I get only `E501` instead... Even for the docstrings...

**Comment by charliermarsh** [comment]
Oh, did you set https://docs.astral.sh/ruff/settings/#lint_pycodestyle_max-doc-length? If that's not set, the rule has no effect (mentioned in the rule docs: https://docs.astral.sh/ruff/rules/doc-line-too-long/).

**Comment by Borda** [comment]
Aha, no... Let me try and thanks for your quick response 🙏
